app/src/test/java/org/simple/clinic/home/overdue/OverdueUiRendererTest.kt

```diff
@@ -124,4 +125,89 @@ class OverdueUiRendererTest {
+        TestData.overdueAppointment(appointmentUuid = UUID.fromString("356099c9-b19d-4be4-8e1e-938eeda8ec66"))
+    )
+    val selectedAppointmentsModel = defaultModel
+        .overdueAppointmentsLoaded(OverdueAppointmentSections(
+            pendingAppointments = pendingAppointments,
+            agreedToVisitAppointments = emptyList(),
+            remindToCallLaterAppointments = emptyList(),
+            removedFromOverdueAppointments = emptyList(),
+            moreThanAnYearOverdueAppointments = emptyList()
+        ))
+        .selectedOverdueAppointmentsChanged(selectedAppointments)
+
+    // when
+    uiRenderer.render(selectedAppointmentsModel)
+
+    // then
+    verify(ui).showOverdueAppointments(
+        OverdueAppointmentSections(
+            pendingAppointments = pendingAppointments,
+            agreedToVisitAppointments = emptyList(),
+            remindToCallLaterAppointments = emptyList(),
+            removedFromOverdueAppointments = emptyList(),
+            moreThanAnYearOverdueAppointments = emptyList()
+        ),
+        selectedOverdueAppointments = selectedAppointments,
+        overdueListSectionStates = selectedAppointmentsModel.overdueListSectionStates
+    )
+    verify(ui).showOverdueCount(2)
+    verify(ui).hideProgress()
+    verify(ui).hideNoOverduePatientsView()
+    verify(ui).showOverdueRecyclerView()
+    verify(ui).showSelectedOverdueAppointmentCount(2)
+    verifyNoMoreInteractions(ui)
+  }
+
+  @Test
+  fun `when no overdue appointments are selected, then hide overdue selected count`() {
+    // given
+    val pendingAppointments = listOf(
+        TestData.overdueAppointment(appointmentUuid = UUID.fromString("801a16dc-9e6c-464c-aa07-993ae4926489")),
+        TestData.overdueAppointment(appointmentUuid = UUID.fromString("356099c9-b19d-4be4-8e1e-938eeda8ec66"))
+    )
+    val selectedAppointmentsModel = defaultModel
+        .overdueAppointmentsLoaded(OverdueAppointmentSections(
+            pendingAppointments = pendingAppointments,
+            agreedToVisitAppointments = emptyList(),
+            remindToCallLaterAppointments = emptyList(),
+            removedFromOverdueAppointments = emptyList(),
+            moreThanAnYearOverdueAppointments = emptyList()
+        ))
+        .selectedOverdueAppointmentsChanged(emptySet())
+
+    // when
+    uiRenderer.render(selectedAppointmentsModel)
+
+    // then
+    verify(ui).showOverdueAppointments(
+        OverdueAppointmentSections(
+            pendingAppointments = pendingAppointments,
+            agreedToVisitAppointments = emptyList(),
+            remindToCallLaterAppointments = emptyList(),
+            removedFromOverdueAppointments = emptyList(),
+            moreThanAnYearOverdueAppointments = emptyList()
+        ),
+        selectedOverdueAppointments = emptySet(),
+        overdueListSectionStates = selectedAppointmentsModel.overdueListSectionStates
+    )
+    verify(ui).showOverdueCount(2)
```

Comment:
The second test is for hiding overdue count although in the test we are not verifying that. We are still verifying the `showOverdueCount` logic. Can we make this change please?

Reply:
That is for verifying hiding selected count. We still have 2 pending appointments, so it should we will still show the count 🤔

Reply:
oohhhh. Stupid me. Sorry

Reply:
Haha, now that there are two different counts, it's a bit confusing 😂